### PR TITLE
Support `chainerx.batch_norm` with 2D input on CUDA

### LINKED ISSUE #6216: ChainerX's 2D batch norm raises only for cuda device
Repro:

```
import numpy as np
import chainerx as chx

with chx.using_device('cuda:0'):
    x = chx.array(np.random.rand(2, 5))
    gamma = chx.array(np.random.rand(5,))
    beta = chx.array(np.random.rand(5,))
    mean = chx.array(np.random.rand(5,))
    var = chx.array(np.random.rand(5,))
    chx.batch_norm(x, gamma, beta, mean, var)
```

Note the above works with `native:0`

### PR DESCRIPTION
**Pull request by beam2d**
Fix #6216. cuDNN batch norm only supports 4 or 5 dimension input, so this PR complements ones to the shape of input descriptors when ndim is lower than them.

### CODE REVIEW

File: tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py [hunk 1]
```diff
@@ -43,10 +43,9 @@ def _create_batch_norm_ndarray_args(
     return x, gamma, beta, mean, var
 
 
-# Note that CUDA (cuDNN) only supports batch normalization with 4 or
-# 5-dimenisional data.
```
Comment (toslunar) [inline tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: Could you keep the note until higher dimensional data are supported by `chainerx.batch_norm`?

Comment (beam2d) [reply tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: That makes sense! Recovered the comment with an additional note.

File: tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py [hunk 2]
```diff
@@ -44,9 +44,11 @@ def _create_batch_norm_ndarray_args(
 
 
 # Note that CUDA (cuDNN) only supports batch normalization with 4 or
-# 5-dimenisional data.
```
Comment (emcastillo) [inline tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: Typo here, `dimenisional`

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Seems like CI failures are related 😞

File: tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py (hunk 2 above)
Comment (beam2d) [reply tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: thx!

**Comment by beam2d** [comment]
Fixed some comments.
I cannot reproduce the CI error... what would be the cause?

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Error seems to be `CUDNN_STATUS_BAD_PARAM ` on ChainerX (not chainer) tests.
In case you can't reproduce might be local cudnn version and CI version missmatch?

**Comment by toslunar** [comment]
See also #8327.

**Comment by toslunar** [comment]
The previous result of Jenkins has gone.

Jenkins, test this please.

**Comment by emcastillo** [comment]
thanks! I will take over it

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by emcastillo** [comment]
I will take over this.

**Comment by emcastillo** [comment]
I was able to reproduce the error using all the jenkins docker images in my local environment.
However, the error changes as I narrow down the test cases and never happens outside docker!🤯

**Comment by emcastillo** [comment]
Takeover by #8464
